install_dependencies_with_nuke.py

```diff
@@ -0,0 +1,40 @@
+"""Installation file for and installing the project dependencies for the nuke interpreter.
+
+Run this script with nuke to install the project dependencies for your nuke interpreter.
+You can execute this script with nuke by running:
+path/to/nuke[.exe] -t install_dependencies_with_nuke.py
+
+Make sure you have a working internet connection.
+"""
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+from Datamodel.constants import NUKE_DEPENDENCY_FOLDER, REQUIREMENT_TXT
+
+
+def install_dependencies() -> None:
```

Comment:
I've thought about this for a while now, I think we could improve the workflow here quite a bit.

How I see it with the NukeTesting package is that we ship it via pip. Its just a normal python package that can be run on the native interpreter thats installed on the system. Or via Nuke, whatever you want.

Then, this package can be invoked from the cli to run the NukeTesting package. A user input is required for the testing folder and the Nuke executable.  However, the user should not have to care about any dependancies at all, besides our pip package.

We then check if all requirements are installed (that are defined by our pip package, so it will be installed automatically when NukeTesting is installed). Then we grab the Nuke major release to check if our dependancies align with the Nuke version. Then we append the pytest library (and other dependancies if needed) to the PATH and invoke the script to run the Nuke Python interpreter. 

If there are incompatibilities, the user should run an older version of NukeTesting that for examples still support Py3.x. However, we try to make it as compatible as possible with all Nuke python versions. Only if really needed we use the newer version of a package.

Anyway, this way we handle all this stuff, and the user can just run tests or use another version of our package if needed.

What do you think? :)

Reply:
I really love your engagement here. I generally agree. The shipping with pip would be awesome. 
I'm just curious why we would need to install it to another python interpreter. 
Something like "nuke -t pip install" would be sufficient IMO.
Of course many nuke users and TDs have python already installed so your point is of course valid. It just adds this extra step of installing a package in an interpreter for another interpreter. 

In regards to this current pr, I'd like to get the first iteration of the test runner done. Once we have a system to run and test our own code with nuke, we are ready for the meat of this project. 
I think we can copy your comment into a new issue because we also need to decide on a releasing mechanism. What du you think?